src/Servers/Kestrel/Core/src/Internal/Http2/Http2FrameWriter.cs

```diff
@@ -372,6 +375,12 @@ public void UpdateMaxFrameSize(int maxFrameSize)
         {
             if (_maxFrameSize != maxFrameSize)
             {
+                // Safe multiply, MaxFrameSize is limited to 2^24-1 bytes by the protocol and by Http2PeerSettings.
+                // Ref: https://datatracker.ietf.org/doc/html/rfc7540#section-4.2
+                if (_maxFrameSize * HeaderBufferSizeMultiplier == _headersEncodingLargeBufferSize)
```

Comment:
I think your goal here was to preserve `_headersEncodingLargeBufferSize` if it had grown beyond its initial value, but I'm not sure that's worthwhile.  I think unconditionally setting it to `maxFrameSize * HeaderBufferSizeMultiplier` is probably fine.  If you'd rather preserve the information, I wonder if it would be preferable/clearer to set it to `Max(_headersEncodingLargeBufferSize, maxFrameSize * HeaderBufferSizeMultiplier)`.

If `_headersEncodingLargeBufferSize` is preserved across max frame size changes, it's possible it will no longer be a multiple of the max frame size.  I'm guessing that won't cause problems?

Reply:
I will remove that. It won't cause a problem.